lib/PrintAsClang/DeclAndTypePrinter.cpp

```diff
@@ -2999,6 +3006,14 @@ bool DeclAndTypePrinter::shouldInclude(const ValueDecl *VD) {
       return false;
   }
 
+  // In C output mode print only @cdecls and skip them in other modes.
```

Comment:
Could you elaborate on this? Where did we emit cdecls before this change?

What if someone what to consume these cdecls from ObjC or C++ code? Is that still possible?

Reply:
We print the new C specific `@cdecl` only in C mode and keep printing the classic `@_cdecl` in other modes only where they would be gated as usual. The previous behavior didn't change for `@_cdecl`. I've updated the comment clarify this.

Objective-C and C++ clients see and can call the newly printed `@cdecl` functions. They are printed in a block at the beginning of the compatibility header without a language check except for an `extern "C"` block.